ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/DataStructureUtil.java

```diff
@@ -300,12 +300,12 @@ public static BeaconBlock randomBeaconBlock(long slotNum) {
         randomBeaconBlockBody());
   }
 
-  public static ArrayList<Deposit> newDeposits(int numDeposits, int slot) {
+  public static ArrayList<Deposit> newDeposits(int numDeposits, int slot, int entropy) {
     ArrayList<Deposit> deposits = new ArrayList<Deposit>();
 
     for (int i = 0; i < numDeposits; i++) {
       // https://github.com/ethereum/eth2.0-specs/blob/0.4.0/specs/validator/0_beacon-chain-validator.md#submit-deposit
-      BLSKeyPair keypair = BLSKeyPair.random(slot + i);
+      BLSKeyPair keypair = BLSKeyPair.random(slot + entropy + i);
```

Comment:
I don't like the addition approach of getting this final entropy, since it might lead to same numbers for different expected validators.

Reply:
Technically true. In reality, slot can be removed since it is only called at Genesis. Since the entropy should be the same for all validators, we would need to make sure that entropy is greater than the num validators specified. That's a long way of saying, I agree

Reply:
i fixed it by repeatedly hashing it rather than adding an index